src/server/routes/controllers/email.js

```diff
@@ -0,0 +1,122 @@
+const config = require('../../config.json')
+const nodemailer = require('nodemailer')
+const logger = require('../../logger')
+const crypto = require('crypto')
+const _ = require('lodash')
+
+const account = config.email
+const EMAIL_SENDER = account.email || process.env.SMTP_USERNAME
+const LOKIM_EMAIL = `"Lokim Messenger Services" <${EMAIL_SENDER}>`
+const INVALID_TOKEN = 'Invalid token.'
+const USER_NOT_FOUND = 'User not found.'
+
+
+const SMTP_OPTIONS = {
+	host: account.hostname || process.env.SMTP_HOSTNAME,
+	port: account.port || process.env.SMTP_PORT,
+	secure: false,
+	auth: {
+		user: EMAIL_SENDER,
+		pass: account.password || process.env.SMTP_PASSWORD
+	}
+}
+
+function prepareTransporter() {
+	let transporter = nodemailer.createTransport(SMTP_OPTIONS)
+	
+	return transporter
+}
+
+function createToken() {
+	const token = crypto.randomBytes(20).toString('hex')
+	return token
+}
+
+function setMailOptions(recvAddress, subject, body) {
+	return {
+		from: LOKIM_EMAIL,
+		to: recvAddress,
+		subject,
+		text: body
+	}
+}
+
+function sendMail(transporter = prepareTransporter()) {
+	return (mailOptions) => {
+		_.forEach(mailOptions, optionValue => {
+			if(_.isEmpty(optionValue)) throw new Error('Mail options cannot be undefined.')
+		})
+
+		transporter.sendMail(mailOptions, (err, info) => {
+			if(err) return logger.warn(`Email failed to send. Details: ${err}`)
+			logger.info('Message sent: %s', info.messageId)
+		})
+	}
+}
+
+function saveRecordToDB(VerifyModel = require('../../models/verification')) {
+	return (username, token) => {
+		if(_.isEmpty(username) || _.isEmpty(token)) throw new Error('You must provide a username and verification token.')
+
+		const verifyData = {username, token}
+		const verifyInstance = new VerifyModel(verifyData)
+        
+		return verifyInstance.save().catch(err => {throw new Error('Error creating Verify record. Please ensure details are unique. More details: ' + err)})//TODO: Handle this better
+	}
+}
+
+function sendVerificationMail(transporter = prepareTransporter()) {
+	return (email, hostname, verificationToken) => {
+		if(_.isEmpty(email) || _.isEmpty(hostname) || _.isEmpty(verificationToken))
+			throw new Error('You must provide a hostname, email and verification token.')
+
+		const link =`${hostname}/verify/${verificationToken}`
+		const subject = 'Verification Email'
+		const body = `You can activate your account at ${link} \nIt will expire after 30  days.`
+
+		sendMail(transporter)(setMailOptions(email, subject, body))
+	}
+}
+
+function verifyUser(
```

Comment:
This function is complicated a little bit. Maybe will be better if this function will be divided into several functions.

Reply:
Refactored the finding of the username out, I don't feel that the remainder of the function is too complex as it currently stands. If you still wish for more changes to this, let me know.

Reply:
@MarcinGrzeszczak Is it fine now?